Src/DependencyCollector/Shared/Implementation/DependencyDiagnosticSourceListener.cs

```diff
@@ -0,0 +1,308 @@
+        {
+            this.client = new TelemetryClient(configuration);
+            this.client.Context.GetInternalContext().SdkVersion = SdkVersionUtils.GetSdkVersion("rdd" + RddSource.DiagnosticSourceExtensibility + ":");
+
+            this.configuration = configuration;
+
+            this.subscriber = new DependencyDiagnosticSourceSubscriber(this);
+        }
+
+        /// <summary>
+        /// Notifies the observer that the provider has finished sending push-based notifications.
+        /// <seealso cref="IObserver{T}.OnCompleted()"/>
+        /// </summary>
+        public void OnCompleted()
+        {
+        }
+
+        /// <summary>
+        /// Notifies the observer that the provider has experienced an error condition.
+        /// <seealso cref="IObserver{T}.OnError(Exception)"/>
+        /// </summary>
+        /// <param name="error">An object that provides additional information about the error.</param>
+        public void OnError(Exception error)
+        {
+        }
+
+        public void OnNext(KeyValuePair<string, object> evnt)
+        {
+            if (evnt.Key.EndsWith(ActivityStopNameSuffix, StringComparison.OrdinalIgnoreCase))
+            {
+                this.OnActivityStop(evnt.Value);
+            }
+        }
+
+        public void Dispose()
+        {
+            if (this.subscriber != null)
+            {
+                this.subscriber.Dispose();
+            }
+        }
+
+        /// <summary>
+        /// Handler for Activity stop event (response is received for the outgoing request).
+        /// </summary>
+        internal void OnActivityStop(object payload)
+        {
+            Activity currentActivity = Activity.Current;
+            if (currentActivity == null)
+            {
+                DependencyCollectorEventSource.Log.CurrentActivityIsNull();
+                return;
+            }
+
+            DependencyCollectorEventSource.Log.DependencyDiagnosticSourceListenerActivityStopped(currentActivity.Id, currentActivity.OperationName);
+
+            DependencyTelemetry telemetry = new DependencyTelemetry();
+
+            // properly fill dependency telemetry operation context
+            telemetry.Context.Operation.Id = currentActivity.RootId;
+            telemetry.Context.Operation.ParentId = currentActivity.ParentId;
+            telemetry.Id = currentActivity.Id;
+            telemetry.Duration = currentActivity.Duration;
+            telemetry.Timestamp = currentActivity.StartTimeUtc;
+
+            foreach (var item in currentActivity.Baggage)
+            {
+                if (!telemetry.Context.Properties.ContainsKey(item.Key))
+                {
+                    telemetry.Context.Properties[item.Key] = item.Value;
+                }
+            }
+
+            this.client.Initialize(telemetry);
+
+            // get the Uri from event payload
+            object uriObject = payload != null ? this.dependencyCallRequestUriFetcher.Fetch(payload) : null;
```

Comment:
I don't like the idea that every DiagnosticsSource should report RequestUri in the payload
1. it's not generic enough: databases have no requests and no Uris
2. it's part of other payloads
3. existing DiagnosticSources does not do it and presumably will not change because of 2